jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -15,6 +17,19 @@ sealed trait Destination
 final case class Topic(name: String) extends Destination
 final case class Queue(name: String) extends Destination
 
+final case class JmsTextMessage(body: String, properties: Map[String, Any] = Map.empty) {
+  def withProperty(name: String, value: Any) = copy(properties = properties + (name -> value))
```

Comment:
I think we in general use `add` as prefix when it does not replace the entire thing it updates but adds to it (in this case it would be `withProperties(map)` which may also make sense for Java users as they can't really use `copy`)

Also, mark it with "Java API"